File: P3VB01AFONSOex/DataBaseAccess.vb
Imports System.Data.SqlClient

Public Class DataBaseAccess
    '################################################################################
    'MÉTODO QUE DEVOLVE A STRING COM A CONEXÃO À BASE DE DADOS
    'nota: pode haver várias SC; escolher qual a que se pretende usar
    Public Function ConnectionString() As String

        'Dim SCantiga As String = "Data Source=121.1............ "
        'Dim SCnova As String = "Data Source=89.1............... "
        'Dim SCLeopardo = "Data Source=leopar...;"
        Dim SCLagostim = "Data Source=10.30.10.2,62555;Initial Catalog=P3VBBD01AFONSO;User ID=EFAafonso;Password=123.Abc##;"
        'Dim SCLagostimExterna = "Data Source=62.28.39.135,62444;Initial Catalog=P3VBBD01AFONSO;User ID=EFAafonso;Password=Abc.123##;"

        'retornar a que é pretendida:
        Return SCLagostim

    End Function
    '################################################################################
    'MÉTODO QUE DEVOLVE A DATABLE A PARTIR DOS PARÂMETROS
    ' SC (string connection) e ssql (query à BD).
    Public Function BuscarDados(SC As String, ssql As String) As DataTable

        Dim Con As SqlConnection = New SqlConnection(SC)
        Con.Open()

        Dim cmd As SqlCommand = New SqlCommand(ssql, Con)

        Dim adapter As SqlDataAdapter = New SqlDataAdapter(cmd)

        Dim dt As New DataTable()
        adapter.Fill(dt)

        adapter.Dispose()
        Con.Close()

        Return dt

    End Function
End Class

File: testeClassesTP/DatabaseAccess.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


    public class DataBaseAccess
    {
        //################################################################################
        //METHOD THAT RETURNS A STRING WITH THE DATABASE CONNECTION
        //note: there may be several SC; choose the one you want to use
        public string ConnectionString()
        {
            string SCLagostimExterna = "Data Source=10.30.10.2,62555;Initial Catalog=exercicioTP;User ID=EFAGONCALVES;Password=123.Abc;";

            //return the one you want to use:
            return SCLagostimExterna;
        }
        //################################################################################
        //METHOD THAT RETURNS A DATATABLE FROM THE PARAMETERS
        // SC (string connection) and ssql (query to the DB).
        public DataTable BuscarDados(string SC, string ssql)
        {
            using SqlConnection Con = new(SC);
            Con.Open();

            using SqlCommand cmd = new(ssql, Con);
            using SqlDataAdapter adapter = new(cmd);
            DataTable dt = new();
            adapter.Fill(dt);

            return dt;
        }
    }
